torch/library.h

```diff
@@ -717,6 +728,20 @@ class TorchLibraryInit final {
   ); \
   void TORCH_LIBRARY_IMPL_init_ ## ns ## _ ## k (torch::Library& m)
 
+// TODO: meaningful comment
+#define TORCH_LIBRARY_IMPL_FRAGMENT(ns, k, m, uid) \
```

Comment:
FRAGMENT is not needed for IMPL, as IMPL already allows multiple blocks for same namespace

Reply:
yes- the issue with using IMPL directly is that it defines a bunch of static variables and gives them names uniquely identified by their namespace. This is a problem for the macro, since it gets called multiple times for different operators, all of which go into the _caffe2:: namespace.

The new macro I added isn't particularly elegant, but it resolves that issue by letting you pass in a unique identifier to avoid name collisions- in this case I'm passing in the particular OperatorName that the macro is registering.

Let me know if that makes sense (and also if you have a cleaner solution- I found usage of a __COUNTER__ macro that looked promising but I wasn't able to get it to work (maybe it's platform dependent).

I also might be able to fold into/replace TORCH_LIBRARY_FRAGMENT_THIS_API_IS_FOR_PER_OP_REGISTRATION_ONLY with TORCH_LIBRARY_FRAGMENT. I just want to dig into how that macro is used a little more first.